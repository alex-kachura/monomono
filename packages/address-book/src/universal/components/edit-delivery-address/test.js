import React from 'react';
import { render } from '@testing-library/react';
import { renderProviders } from '@oneaccount/react-foundations/lib/test-utils';
import { DefaultThemeProvider } from '@beans/theme';
import EditDeliveryAddressPage from './';

function renderPage() {
  const mockLocation = {
    pathname: 'mock-path',
    search: 'mock-search',
  };

  return render(
    renderProviders({
      appConfig: {
        getLocalePhrase: (key) => key,
        csrf: 'super-secure-token',
      },
      initialPageData: {
        fields: [],
        schema: {},
        errors: {},
        values: {},
      },
      children: (
        <DefaultThemeProvider>
          <EditDeliveryAddressPage location={mockLocation} />
        </DefaultThemeProvider>
      ),
    }),
  );
}

describe('[Page: EditDeliveryAddressPage]', () => {
  describe('[State: Default]', () => {
    it('should render correctly', () => {
      const component = renderPage();

      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
