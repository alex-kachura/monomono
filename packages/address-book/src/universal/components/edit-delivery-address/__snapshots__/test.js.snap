// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Page: EditDeliveryAddressPage] [State: Default] should render correctly 1`] = `
<DocumentFragment>
  <h1
    class="sc-htoDjs krCBJg typography___styledElement-sc-186l0ce-0 hmbiuS"
  >
    pages.delivery-address.edit.title
  </h1>
  <div
    class="sc-bZQynM gBCjJB"
  >
    <form
      autocomplete="off"
      method="POST"
      novalidate=""
    >
      <input
        name="_csrf"
        type="hidden"
        value="super-secure-token"
      />
      <button
        aria-disabled="false"
        class="sc-iwsKbI coJpLO styled__PrimaryButton-sc-3mnirm-2 giYBMC"
        role="button"
        type="submit"
      >
        <span
          class="styled__InnerContainer-sc-3mnirm-1 fUwmia"
        >
          <span
            class="beans-button__text styled__TextSpan-sc-3mnirm-5 eqKhmg"
          >
            pages.delivery-address.edit.submit-button
          </span>
        </span>
      </button>
    </form>
  </div>
</DocumentFragment>
`;
